services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DelegatingDeviceManagementHttpEndpoint.java

```diff
@@ -244,4 +282,46 @@ private void doDeleteDevice(final RoutingContext ctx) {
                     () -> result.complete(new Device()));
         return result.future();
     }
+
+    private <T> Future<Optional<List<T>>> decodeJsonFromRequestParameter(final RoutingContext ctx,
+            final String paramKey, final Class<T> clazz) {
+
+        Objects.requireNonNull(ctx);
+        Objects.requireNonNull(paramKey);
+        Objects.requireNonNull(clazz);
+
+        final Promise<Optional<List<T>>> result = Promise.promise();
+        try {
+            final Optional<List<T>> values = Optional.ofNullable(ctx.request().params()
+                    .getAll(paramKey))
+                    .map(jsons -> jsons
+                            .stream()
+                            .map(json -> Json.decodeValue(json, clazz))
+                            .collect(Collectors.toList()));
+            result.complete(values);
+        } catch (final DecodeException e) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                    String.format("error parsing json value of parameter [%s]", paramKey), e));
+        }
+
+        return result.future();
+    }
+
+    private Future<Optional<Integer>> getRequestParameterIntegerValue(final RoutingContext ctx,
+            final String paramKey) {
+        Objects.requireNonNull(ctx);
+        Objects.requireNonNull(paramKey);
+
+        final Promise<Optional<Integer>> result = Promise.promise();
+        try {
+            final Optional<Integer> value = Optional.ofNullable(ctx.request().params().get(paramKey))
+                    .map(Integer::parseInt);
+            result.complete(value);
```

Comment:
what about checking the min/max values and returning a default value?

Reply:
The properties such as `tenantId` and `deviceId` patterns used for validations are made configurable by adding them to the `ServiceConfigProperties`. Following the same approach, _min_, _max_ and _default_ values for page size and offset could be made configurable. FMPOV these properties are specific to the device registry and doesn't suit well in that class. Hence I have created a new class `DeviceRegistryManagementHttpServiceConfigProperties` which extends `HttpServiceConfigProperties` to accommodate these _min_, _max_ and _default_ config properties. I have pushed a separate commit with those changes. If there is another alternate better way to achieve the above, I am open for it :)

Reply:
FMPOV there is no need to make these configurable as we have defined all those values in the OpenAPI spec ...